src/sisl_toolbox/siesta/minimizer/basis_optimization.py

```diff
@@ -220,18 +225,20 @@ def generate_basis_config(
         X can also be S, D, T, Q for 1, 2, 3, 4 zetas respectively. E.g. "DZ".
 
         It can also be "atoms", which exactly takes the orbitals present in the atoms object.
-    optimize_pol : bool, optional
-        If the basis contains polarization orbitals, whether they should be
-        explicitly optimized or not. If not, the default polarization orbitals
-        will be requested.
-
-        The polarization orbitals are optimized through the charge confinement.
-    pol_optimize_screening : bool, optional
-        If polarization orbitals are optimized, whether the screening length
-        of the charge confinement should be optimized or not.
-    pol_optimize_width : bool, optional
-        If polarization orbitals are optimized, whether the width of the charge confinement
-        should be optimized or not.
+    optimize_charge_conf : bool, tuple of bool, optional
+        Whether some orbitals will be optimized for charge-confinement.
+        If true, it will optimize, the charge (Q), the Yukawa screening parameter
+        :math:`\lambda`, and the singularity regularization parameter :math:`\delta`.
+
+        For multiple arguments, each can be fine-tuned on/off.
+
+        This will only optimize charge-confinement for *empty* orbitals (:math:`q=0`).
+    optimize_soft_conf : bool, tuple of bool, optional
+        Whether all orbitals will be optimized for soft-confinement.
+        If true, it will optimize, the potential (:math:`V_0`), and the inner
+        radius :math:`r_i`.
+
+        For multiple arguments, each can be fine-tuned on/off.
```

Comment:
Does this work from the CLI? If so, how?

Doesn't it make the CLI harder to understand? 
`--optimize-soft-conf false` will optimize soft confinement.

For the charge confinement, are you sure optimizing the three parameters at the same time will result in a good optimization/ is worth it? I also don't understand to which orbitals it applies, Federico said that the charge optimization should only be done for polarization orbitals. So again make sure to talk to Federico because I am not sure that this added flexibility is good.